docs/orchestration.asciidoc

```diff
@@ -0,0 +1,104 @@
+
+[id="{p}-upgrading"]
+==== Upgrading the cluster
+
+ECK handles smooth upgrades from one cluster specification to another.
+
+Some examples based on the Elasticsearch specification above:
+
+- To add 5 additional data nodes, simply change `count: 10` to `count: 15` in the `data-nodes` NodeSet, then apply the Elasticsearch resource.
+- To increase the RAM memory limit of data Nodes to 32Gi, link:k8s-managing-compute-resources.html[set a different resources limits] in the existing `data-nodes` NodeSet PodTemplate.
+- To replace dedicated master and dedicated data nodes by nodes having both master and data roles, replace the 2 existing NodeSets by a single one with a different name and the corresponding Elasticsearch configuration settings.
+
+No downtime is expected when NodeSet changes are orchestrated by ECK. It ensures that:
+
+- data from a node to remove is completely migrated away to other nodes before the node is actually removed
+- Elasticsearch orchestration settings such as `discovery.seed_hosts`, `cluster.initial_master_nodes`, `discovery.zen.minimum_master_nodes`, `_cluster/voting_config_exclusions` correctly account for any change in the cluster
+- rolling upgrades are performed in a safe way, reusing PersistentVolumes of the upgraded Elasticsearch nodes
+
+[id="{p}-statefulsets"]
+==== StatefulSets orchestration
+
+Behind the scenes, ECK translates each NodeSet specified in the Elasticsearch resource into a link:https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/[StatefulSet in Kubernetes]. The StatefulSet specification is based on the NodeSet specification:
+
+* `count` corresponds to the number of replicas in the StatefulSet, each replica leading to the creation of a Pod, which corresponds to a single Elasticsearch node
+* `podTemplate` is patched with additional required settings to run an Elasticsearch Pod, including the Elasticsearch container and sane default settings for the Pod
+* the StatefulSet name is built from the Elasticsearch resource name and the NodeSet name (`clustername-nodesetname`). Each Pod will be assigned the StatefulSet name, suffixed by an ordinal (`clustername-nodesetname-0`). The corresponding Elasticsearch node has the same name.
+
+The actual Pod creation is handled by the StatefulSet controller in Kubernetes. However, ECK sets the link:https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#on-delete[StatefulSet update strategy] to `OnDelete`, since it needs full control over when and how Pods get upgraded to a new revision.
+
+Whenever a Pod is removed and recreated (maybe with a newer revision), the StatefulSet controller ensures PersistentVolumes that were attached to the original Pod are attached again to the replacing Pod.
+
+[id="{p}-upgrade-patterns"]
+==== Cluster upgrade patterns
+
+Depending on how the NodeSets specification are updated, ECK handles the Kubernetes resources reconciliation in various ways.
```

Comment:
Either `NodeSets specifications` or `is updated`

Reply:
Maybe "Depending on how the NodeSets are updated"?